gui/static/js/components/addbot.js

```diff
@@ -0,0 +1,171 @@
+var React = require('react');
+
+class DiscoveredBot extends React.Component {
+    render() {
+        var styles = {
+            ipAddress: {
+                float: "left"
+            }
+        }
+        return (
+            <div className="discoveredbot">
+                <p style={styles.ipAddress}>{this.props.ip_address}</p>
+                <button id={'discoverBot' + this.props.idx} value={this.props.ip_address} className="addBot">
+                    add bot
+                </button>
+            </div>
+        )
+    }
+}
+
+export default class AddBot extends React.Component {
+    //TODO
+    constructor(props) {
+        super(props);
+        this.state = {
+            ip: "",
+            port: "10000",
+            name: "Bot0",
+            type: "minibot",
+            discoveredBots: []
+        };
+
+        this.handleInputChange = this.handleInputChange.bind(this);
+        this.addbot = this.addbot.bind(this);
+        this.updateDiscoveredBots = this.updateDiscoveredBots.bind(this);
+    }
+
+    handleInputChange(event) {
```

Comment:
Nitpicky, but please provide a short jsdoc description of what each function does, here and below.

ex)
```JavaScript
/**
 * Deals with changes in input. Assumes [x, y, z].
 */
handleInputChange(event) {
  // Contents
}
```